### PR TITLE
Tweak TeamMemberRole.php so that it does not cause errors with the JS…

### PR DESCRIPTION
**Pull request by audave**
…ONAPI
https://github.com/apigee/apigee-edge-drupal/issues/845

### CODE REVIEW

File: modules/apigee_edge_teams/src/Entity/TeamMemberRole.php [hunk 1]
```diff
@@ -56,6 +60,31 @@ final class TeamMemberRole extends ContentEntityBase implements TeamMemberRoleIn
 
   use EntityChangedTrait;
 
+  /**
+   * {@inheritdoc}
+   */
+  public function access($operation, AccountInterface $account = NULL, $return_as_object = FALSE) {
+    if (!$account) {
+      // If for we have hit this without an account return forbidden.
```
Comment (mxr576) [inline modules/apigee_edge_teams/src/Entity/TeamMemberRole.php]: Unnecessary "for"

File: modules/apigee_edge_teams/src/Entity/TeamMemberRole.php [hunk 2]
```diff
@@ -56,6 +60,31 @@ final class TeamMemberRole extends ContentEntityBase implements TeamMemberRoleIn
...
+
+    $result = AccessResult::allowedIfHasPermissions($account, [
+      'administer team',
+      'manage team members',
+    ], 'OR');
+
+    if ($result->isNeutral()) {
+      $team = $this->getTeam();
+      $team_permission_handler = \Drupal::service('apigee_edge_teams.team_permissions');
+      $result = AccessResultAllowed::allowedIf(in_array('team_manage_members', $team_permission_handler->getDeveloperPermissionsByTeam($team, $account)))
```
Comment (mxr576) [inline modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: The $result created in line 72 should be concatenated with this new one by using or to preserve cacheability information originating from that result object.

Comment (shishir-intelli) [reply modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: Hi @audave,
I agree with @mxr576 , 
Could you merge the `$result` on line 72 with the new one, so we can review this PR and approve it.

Also, could you please verify your CLA which is failing.

Comment (audave) [reply modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: allowedIf method does not take a conjunction.

The method used here matches the implementation in TeamAppListByTeamAccess->access()

Comment (shishir-intelli) [reply modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: @audave 
Could you please sign the Google cla.
Ref [here](https://github.com/apigee/apigee-edge-drupal/pull/848/checks)

**Comment by shishir-intelli** [comment]
@audave 
Could you please sign the Google CLA.
Ref [here](https://github.com/apigee/apigee-edge-drupal/pull/848/checks)

File: modules/apigee_edge_teams/src/Entity/TeamMemberRole.php (hunk 2 above)
Comment (shishir-intelli) [reply modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: > allowedIf method does not take a conjunction.
> 
> The method used here matches the implementation in TeamAppListByTeamAccess->access()

@audave 
Could you try to concatenate like this
```
return AccessResult::allowedIfHasPermissions($account, [
    'administer team',
    'manage team members',
    ], 'OR')
    ->orIf(AccessResultAllowed::allowedIf(in_array('team_manage_members', $team_permission_handler->getDeveloperPermissionsByTeam($team, $account))))
    ->addCacheableDependency($team)
    ->cachePerUser();
```

Comment (shishir-intelli) [reply modules/apigee_edge_teams/src/Entity/TeamMemberRole.php:80]: @audave We will create separate PR for above code change.

**Review by shishir-intelli — APPROVED** [review]
LGTM

**Pull request merged**